app/models/ability.rb

```diff
@@ -0,0 +1,15 @@
+# # frozen_string_literal: true
+
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    can :read, :all if user.nil?
```

Comment:
[ OPTIONAL] 
- A better way would have been to create a `user.admin?` method. See sample code snippet below:
    
```
def initialize(user)

         user ||= User.new # guest user (not logged in)

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end
     end

```
In the user model, then you have the method below:
...
```
   def admin?
     role == 'admin' ? true : false
   end
```

Reply:
- Thanks for your hint I will try to make that to benefit from you 
### Thanks again and have a good day 😊